components/script/script_module.rs

```diff
@@ -680,6 +680,34 @@ impl ModuleTree {
         ServoUrl::parse_with_base(Some(url), &specifier_str)
     }
 
+    /// <https://html.spec.whatwg.org/multipage/#resolving-a-url-like-module-specifier>
+    #[allow(unsafe_code)]
+    fn resolve_url_like_module_specifier(
```

Comment:
Let's call this method in `resolve_module_specifier`. Looking to the usage of `resolve_module_specifier`, I think it's also safe to change it to return `Option`.